### PR TITLE
feat (IAM Access Groups): Add support for access group templates

### PR DESCRIPTION
**Pull request by hariarla**
## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## Current vs new behavior  

SDK adopters will be able to call access group template and assignment APIs.

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

## Other information
API Definition
[Stage](https://test.cloud.ibm.com/docs/access-groups)
[Production](https://cloud.ibm.com/docs/access-groups)

Tests:
Screenshot 2023-07-24 at 10 35 18 AM
Screenshot 2023-07-24 at 10 34 49 AM

Examples:
Screenshot 2023-07-24 at 10 50 33 AM

Integration Tests:
Screenshot 2023-07-24 at 10 39 58 AM

### CODE REVIEW

File: examples/iam-access-groups.v2.test.js [hunk 1]
```diff
@@ -664,4 +692,727 @@ describe('IamAccessGroupsV2', () => {
...
+      originalLog(output);
+    });
+    consoleWarnMock.mockImplementation((output) => {
+      // if an error occurs, display the message and then fail the test
+      originalWarn(output);
+      expect(true).toBeFalsy();
+    });
+
+    originalLog('getTemplateVersion() result:');
+    // begin-get_template_version
```
Comment (padamstx) [inline examples/iam-access-groups.v2.test.js:1117]: Limited to only 1 code example per operation.

**Review by padamstx — CHANGES_REQUESTED** [review]
Just one issue that I noticed... code examples can have only 1 code block per operation.

File: examples/iam-access-groups.v2.test.js (hunk 1 above)
Comment (hariarla) [reply examples/iam-access-groups.v2.test.js]: fixed, and test results updated.

**Comment by hariarla** [comment]
Tests:
Screenshot 2023-07-25 at 1 24 57 AM
Screenshot 2023-07-25 at 10 37 59 AM

Integration Tests:
Screenshot 2023-07-25 at 10 58 16 AM


Example Tests:
Screenshot 2023-07-25 at 6 36 43 PM

**Review by pyrooka — APPROVED** [review]
LGTM!

**Review by padamstx — APPROVED** [review]
LGTM

**Comment by hariarla** [comment]
Tests:
Screenshot 2023-08-21 at 1 04 02 AM
Screenshot 2023-08-21 at 12 57 35 AM
Screenshot 2023-08-21 at 12 47 17 AM
Screenshot 2023-08-21 at 12 41 09 AM

**Pull request merged**